### PR TITLE
fix(redmine): Handle required fields

### PR DESCRIPTION
**Pull request by mikellykels**
Handle the required fields project_id, tracker_id, and default_priority after Host and Key credentials have been added.

## Before

Even when Host/Key was saved, the other required fields (Project, Tracker, and Default Priority) would not populate with the required information in order to continue:
screen shot 2018-07-03 at 2 46 36 am

## After

Enter Host/Key > Save:
screen shot 2018-07-03 at 1 10 41 am

Continue to connect/Save your Project, Tracker, Default Priority, and any Extra Fields:
screen shot 2018-07-03 at 1 12 44 am

Error messages added:
screen shot 2018-07-03 at 1 11 58 am

### CODE REVIEW

File: sentry_redmine/plugin.py [hunk 1]
```diff
@@ -205,18 +206,31 @@ def get_config(self, project, **kwargs):
...
                 self.remove_field(field_name)
-        
+
         return self.fields
-    
+
 
     def validate_config(self, project, config, actor):
         super(RedminePlugin, self).validate_config(project, config, actor)
+        self.client_errors = []
```
Comment (dcramer) [inline sentry_redmine/plugin.py:219]: A better approach here would be to coerce the fields to a map so you’re not reliant on order

Fields = {f[‘name’]: f for f in self.fields}

You could even just create a list of the required fields already being their config using something similar with list comprehension 

Pardon syntax I’m on my phone

Comment (mikellykels) [reply sentry_redmine/plugin.py:219]: talked with colleen some on this too and refactored the code a little differently, but let me know your thoughts.

**Review by dcramer — APPROVED** [review]
In general looks like a step in the right direction.

One more "more correct" followup you could do is validating each field as needed using the ``validate_config_field`` hook (which I THINK is available here).

You'll see that used in project_plugin_details.py, and in that case you can basically raise a ``serializers.ValidationError('reason this field did not validate')`` and itll automatically populate to the right fields in the UI.

**Comment by mikellykels** [comment]
I created another JIRA ticket for further improvements where I can revisit improving/refactoring this integration https://getsentry.atlassian.net/browse/SE-83 (`serializers.ValidationError` may need a little more time to get working). Going to merge this now so people can use the integration.

**Pull request merged**